dask/dot.py

```diff
@@ -1,99 +1,74 @@
 from __future__ import absolute_import, division, print_function
 
-import networkx as nx
-from dask.core import istask, get_dependencies
+import os
 
+from graphviz import Digraph
 
-def make_hashable(x):
-    try:
-        hash(x)
-        assert len(str(x)) < 100
-        return x
-    except:
-        return hash(str(x))
-
+from .core import istask, get_dependencies, ishashable
 
-def lower(func):
-    while hasattr(func, 'func'):
-        func = func.func
-    return func
 
-def name(func):
+def label(func):
     try:
-        return lower(func).__name__
+        while hasattr(func, 'func'):
+            func = func.func
+        return func.__name__
     except AttributeError:
         return 'func'
 
 
-def to_networkx(d, data_attributes=None, function_attributes=None):
+def name(x):
+    try:
+        return str(hash(x))
+    except TypeError:
+        return str(hash(str(x)))
```

Comment:
OK.  I suppose my concerns would be mollified if we were to improve node labeling, particularly for complex tasks like 

    (f, x, (g, y, 10))

Naming this as just `f` loses a lot of the meaning.  

I wonder if we could effectively names these nodes as follows:

    f(-, g(-, 10))

This is related to #413 I guess and could be useful in a few places.